### PR TITLE
create prometheus addl config for prometheus-operator

### PR DESCRIPTION
**Pull request by kanagarajkm**
- Removed ServiceMonitor. Using ServiceMonitor leads to prometheus scrapping all the pods for the cluster level metrics which is not desired
- Additional scrape config will be added to `Prometheus` resource when `prometheusOperator` is enabled for a tenant

### CODE REVIEW

File: pkg/controller/cluster/main-controller.go [hunk 1]
```diff
@@ -1267,16 +1257,14 @@ func (c *Controller) syncHandler(key string) error {
 	}
 
 	if tenant.HasPrometheusSMEnabled() {
-		err = c.checkAndCreatePrometheusServiceMonitorSecret(ctx, tenant, string(tenantConfiguration["accesskey"]), string(tenantConfiguration["secretkey"]))
-		if err != nil {
-			return err
-		}
-		err = c.checkAndCreatePrometheusServiceMonitor(ctx, tenant)
+		klog.Infof("Adding Prometheus addl config")
```
Comment (harshavardhana) [inline pkg/controller/cluster/main-controller.go:1260]:
```suggestion
		klog.Infof("Adding MinIO tenant Prometheus scrape config")
```

File: pkg/controller/cluster/main-controller.go [hunk 2]
```diff
@@ -1267,16 +1257,14 @@ func (c *Controller) syncHandler(key string) error {
...
+		err := c.checkAndCreatePrometheusAddlConfig(ctx, tenant, string(tenantConfiguration["accesskey"]), string(tenantConfiguration["secretkey"]))
 		if err != nil {
 			return err
 		}
 	} else {
-		err = c.deletePrometheusServiceMonitor(ctx, tenant)
+		klog.Infof("Deleting Prometheus addl config")
```
Comment (harshavardhana) [inline pkg/controller/cluster/main-controller.go:1266]:
```suggestion
		klog.Infof("Deleting MinIO tenant Prometheus scrape config"")
```

File: resources/base/cluster-role.yaml [hunk 3]
```diff
@@ -118,6 +118,7 @@ rules:
       - monitoring.coreos.com
     resources:
       - servicemonitors
```
Comment (harshavardhana) [inline resources/base/cluster-role.yaml:120]: We need to remove 

```suggestion
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]
This would require CRD updates @kanagarajkm - don't see them in the PR?

**Comment by kanagarajkm** [comment]
@harshavardhana i have updated the CRD, please take a look.

**Review by harshavardhana — APPROVED** [review]
LGTM haven't tested yet

**Review by dvaldivia — DISMISSED** [review]

**Comment by harshavardhana** [comment]
@kanagarajkm can you fix the conflicts?

**Comment by kanagarajkm** [comment]
@harshavardhana i have fixed the conflicts, PTAL

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
PTAL @dvaldivia

**Review by dvaldivia — APPROVED** [review]

**Comment by fionera** [comment]
Why was this done instead of modifying the scrape config to only scrape one of the instances for these metrics. The new approach does not work with any other setup except one case where you have only a prometheus in a specific namespace. As I understand it I cannot have a seperate prometheus instance per tenant and support for using PrometheusAgent is completely missing

**Comment by chriskhanhtran** [comment]
> Why was this done instead of modifying the scrape config to only scrape one of the instances for these metrics. The new approach does not work with any other setup except one case where you have only a prometheus in a specific namespace. As I understand it I cannot have a seperate prometheus instance per tenant and support for using PrometheusAgent is completely missing

@fionera 
> Starting with prometheus-operator v0.65.x, one can use the ScrapeConfig CRD to scrape targets external to the Kubernetes cluster or create scrape configurations that are not possible with the higher level ServiceMonitor/Probe/PodMonitor resources. (source: https://prometheus-operator.dev/docs/developer/scrapeconfig/)

I think creating ScrapeConfig CR is a better and neater way to add scrape config, instead of using MinIO operator to modify Prometheus CR, which could be managed by Helm/Terraform.

**Pull request merged**